apps/fluent-tester/src/E2E/Menu/pages/MenuPageObject.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  MENU_TESTPAGE,
+  MENUTRIGGER_TEST_COMPONENT,
+  MENUITEM_NO_A11Y_LABEL_COMPONENT,
+  HOMEPAGE_MENU_BUTTON,
+  MENU_ON_OPEN,
+  MENU_ON_CLOSE,
+  MENUITEM_TEST_COMPONENT,
+  MENUPOPOVER_TEST_COMPONENT,
+} from '../../../FluentTester/TestComponents/Menu/consts';
+import { BasePage, By } from '../../common/BasePage.win';
+
+/* This enum gives the spec file an EASY way to interact with SPECIFIC UI elements on the page.
+ * The spec file should import this enum and use it when wanting to interact with different elements on the page. */
+export const enum MenuComponentSelector {
+  PrimaryComponent, //this._primaryComponent
+  SecondaryComponent, //this._secondaryComponent
+  TertiaryComponent, //this._tertiaryComponent
+}
+
+class MenuPageObject extends BasePage {
+  /******************************************************************/
+  /**************** UI Element Interaction Methods ******************/
+  /******************************************************************/
+  didMenuOpen(): boolean {
+    const callbackText = By(MENU_ON_OPEN);
+    browser.waitUntil(
+      () => {
+        return callbackText.isDisplayed();
+      },
+      {
+        timeout: this.waitForPageTimeout,
+        timeoutMsg: 'The Menu did not open.',
+        interval: 1000,
+      },
+    );
+
+    return callbackText.isDisplayed();
+  }
+
+  didMenuClose(): boolean {
+    const callbackText = By(MENU_ON_CLOSE);
+    browser.waitUntil(
+      () => {
+        return callbackText.isDisplayed();
+      },
+      {
+        timeout: this.waitForPageTimeout,
+        timeoutMsg: 'The Menu did not close.',
+        interval: 1000,
+      },
+    );
+
+    return callbackText.isDisplayed();
+  }
+
+  getMenuItemAccessibilityLabel(componentSelector: MenuComponentSelector): string {
```

Comment:
Also, just an idea for future :) I added accessibility tests today the same way but it contains a lot of repeating logic. As these are common methods for accessibility testing, maybe it would be good to add in future to BasePage.win accessibilityMathods which can take sort of ...args for selectors. Because currently in BasePage.win ComponentSelector contains only two selectors and getAccessibilityRole method gets only one selector as argument. In majority of cases we need to add new logic which does the same